engine/class_modules/sc_death_knight.cpp

```diff
@@ -6649,6 +6665,88 @@ struct summon_gargoyle_t : public death_knight_spell_t
   }
 };
 
+// Swarming Mist ============================================================
+
+struct swarming_mist_damage_t : public death_knight_spell_t
+{
+  int swarming_mist_energize_target_cap;
+  int swarming_mist_energize_tick;
+
+  swarming_mist_damage_t( death_knight_t* p ) :
+    death_knight_spell_t( "swarming_mist_damage", p, p -> covenant.swarming_mist -> effectN( 1 ).trigger() ),
+    swarming_mist_energize_target_cap( p -> covenant.swarming_mist -> effectN( 3 ).base_value() ),
+    swarming_mist_energize_tick( 0 )
+  {
+    background = true;
+    aoe = -1;
+    base_multiplier *= 1.0 + p -> conduits.impenetrable_gloom.percent();
+  }
+
+  void execute() override
+  {
+    swarming_mist_energize_tick = 0;
+    death_knight_spell_t::execute();
+  }
+
+  void impact( action_state_t* s ) override
+  {
+    death_knight_spell_t::impact( s );
+    if ( swarming_mist_energize_tick < swarming_mist_energize_target_cap )
+    {
+      p() -> resource_gain( RESOURCE_RUNIC_POWER,
+                 p() -> covenant.swarming_mist_energize->effectN( 1 ).resource( RESOURCE_RUNIC_POWER ),
+                 p() -> gains.swarming_mist );
+      swarming_mist_energize_tick++;
+    }
+  }
+};
+
+struct swarming_mist_buff_t : public buff_t
+{
+  swarming_mist_damage_t* damage; // (AOE) damage that ticks every second
+
+  swarming_mist_buff_t( death_knight_t* p ) :
+    buff_t( p, "swarming_mist", p -> covenant.swarming_mist ),
+    damage( new swarming_mist_damage_t( p ) )
+  {
+    cooldown -> duration = 0_ms; // Controlled by the action
+    set_refresh_behavior( buff_refresh_behavior::DURATION );
```

Comment:
Is it even able to refresh? I wouldn't touch the default value unless necessary (though refresh behavior shouldn't ever be used anyway)

Reply:
Duration isn't long enough to refresh.  NGL it was just a straight copy of remorseless.  I had wondered about leaving that in.